src/lib/api/fetchRSS.ts

```diff
@@ -18,108 +13,123 @@ export const fetchRSS = async (xmlUrl: string | string[]) => {
+          })
+          // Sort by published (most recent is first in array
+          .sort((a, b) => {
+            const dateA = new Date(a.updated[0])
+            const dateB = new Date(b.updated[0])
+            return dateB.getTime() - dateA.getTime()
+          })
+          // Map to RSSItem object
+          .map((entry) => {
+            const getString = (el?: AtomElement[]): string => {
+              if (!el) return ""
+              const [firstEl] = el
+              if (typeof firstEl === "string") return firstEl
+              return firstEl._ || ""
+            }
+            const getHref = (): string => {
+              if (!entry.link) {
+                console.warn(`No link found for RSS url: ${url}`)
+                return ""
+              }
+              const link = entry.link[0]
+              if (typeof link === "string") return link
+              return link.$.href || ""
+            }
+            const getImgSrc = (): string => {
+              const imgRegEx = /https?:\/\/[^"]*?\.(jpe?g|png|webp)/g
+              const contentMatch = getString(entry.content).match(imgRegEx)
+              if (contentMatch) return contentMatch[0]
+              const summaryMatch = getString(entry.summary).match(imgRegEx)
+              if (summaryMatch) return summaryMatch[0]
+              return feedImage || ""
+            }
+            return {
+              pubDate: entry.updated[0],
+              title: getString(entry.title),
+              link: getHref(),
+              imgSrc: getImgSrc(),
+              source,
+              sourceUrl,
+              sourceFeedUrl: url,
             }
-            const link = entry.link[0]
-            if (typeof link === "string") return link
-            return link.$.href || ""
-          }
-          const getImgSrc = (): string => {
-            const imgRegEx = /https?:\/\/[^"]*?\.(jpe?g|png|webp)/g
-            const contentMatch = getString(entry.content).match(imgRegEx)
-            if (contentMatch) return contentMatch[0]
-            const summaryMatch = getString(entry.summary).match(imgRegEx)
-            if (summaryMatch) return summaryMatch[0]
-            return feedImage || ""
-          }
-          return {
-            pubDate: entry.updated[0],
-            title: getString(entry.title),
-            link: getHref(),
-            imgSrc: getImgSrc(),
-            source,
-            sourceUrl,
-            sourceFeedUrl: url,
-          } as RSSItem
-        })
+          })
 
-      allItems.push(parsedAtomItems)
+        allItems.push(parsedAtomItems)
+      } else {
+        throw new Error(
+          `Error parsing XML, invalid RSSResult or AtomResult type: ${url}`
+        )
+      }
+    } catch (error) {
+      console.error(error instanceof Error ? error.message : error)
+      continue
     }
   }
-  return allItems as RSSItem[][]
+
+  if (allItems.length < RSS_DISPLAY_COUNT)
+    throw new Error("Insufficient number of RSS items fetched")
```

Comment:
This isn't actually breaking the build as-is, since it is contained inside an api function call... 🤔  cc: @pettinarip